project-base/storefront/components/Layout/Header/MobileMenu/MobileMenu.tsx

```diff
@@ -0,0 +1,74 @@
+import { DropdownMenuListItem } from './MobileMenuListItem';
+import { SubMenu } from './MobileMenuSubItems';
+import { mapNavigationMenuItems } from './helpers';
+import { ArrowIcon, CloseIcon } from 'components/Basic/Icon/IconsSvg';
+import { NavigationQueryApi } from 'graphql/generated';
+import useTranslation from 'next-translate/useTranslation';
+import { useState } from 'react';
+
+export type MenuItem = {
+    name: string;
+    link: string;
+    parentItem?: string;
+    children?: MenuItem[];
+};
+
+type MobileMenuProps = {
+    navigationItems: NavigationQueryApi['navigation'];
+    onMenuToggleHandler: () => void;
+};
+
+export const MobileMenu: FC<MobileMenuProps> = ({ navigationItems, onMenuToggleHandler }) => {
+    const { t } = useTranslation();
+    const [historyMenuGroups, setHistoryMenuGroups] = useState<MenuItem[][] | undefined>();
+    const [currentMenuItems, setCurrentMenuItems] = useState<MenuItem[]>(mapNavigationMenuItems(navigationItems));
+
+    const currentGroupTitle = currentMenuItems[0].parentItem;
+
+    return (
+        <>
+            <div className="py-3 flex">
+                {!!historyMenuGroups?.length && (
+                    <button
+                        className="cursor-pointer flex w-9 justify-start gap-2 items-center text-sm uppercase text-dark"
+                        title={t('Back')}
+                        onClick={() => {
+                            setCurrentMenuItems([...historyMenuGroups].pop()!);
+                            setHistoryMenuGroups(historyMenuGroups.slice(0, -1));
+                        }}
+                    >
+                        <ArrowIcon className="rotate-90 w-5" />
+                    </button>
+                )}
+
+                {currentGroupTitle && (
+                    <span className="text-center flex-1 leading-5 uppercase">{currentGroupTitle}</span>
+                )}
+
+                <button
+                    className="cursor-pointer ml-auto flex w-9 justify-end gap-2 items-center text-sm uppercase text-dark"
+                    title={t('Close')}
+                    onClick={onMenuToggleHandler}
+                >
+                    <CloseIcon className="w-5" />
+                </button>
+            </div>
+
+            <div>
+                {currentMenuItems.map((navigationItem) => (
+                    <DropdownMenuListItem
+                        key={navigationItem.link}
+                        navigationItem={navigationItem}
+                        onNavigate={onMenuToggleHandler}
+                        onExpand={() => {
+                            setHistoryMenuGroups([...(historyMenuGroups || []), currentMenuItems]);
```

Comment:
Same here. Maybe it could use a different syntax than spreading, but I will leave it to you.

Reply:
Here I don't see any other option, if I don't want to modify original value. Which I shouldn't since it is returned from `useState`.